### DISCUSSION TITLE
Input loosing focus after table rerender

### DISCUSSION THREAD

**Discussion by happylolonly** [TanStack/table#5572] [Q&A]
How to keep focus on input when table (columns) rerenders?
Tried `key` on input, but it is not working...

**Comment by Jarzka** [comment]
I have had similar problems and on my case, all of them were related to the fact the some part of my component tree was re-mounted when an user typed something. If it wasn't the input component itself, it was some of its parent components. Make sure that React key is present everywhere where you create components via looping.

**Comment by AndrejNemec** [comment]
Try wrap input with React.memo

**Reply by AndrejNemec** [reply]
working [example](https://codesandbox.io/p/devbox/jolly-boyd-ct9s4g?workspaceId=079fe425-5eb0-420c-a5ee-a220ef4aed14&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clwj1vvib00083b6gjgqw5hba%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clwj1vvib00033b6gux9tr8v5%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clwj1vvib00053b6glfdsldgw%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clwj1vvib00073b6gi11go318%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clwj1vvib00033b6gux9tr8v5%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clwj1vvib00023b6g0eatr4cs%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Fmain.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A243%252C%2522startColumn%2522%253A2%252C%2522endLineNumber%2522%253A243%252C%2522endColumn%2522%253A2%257D%255D%257D%255D%252C%2522id%2522%253A%2522clwj1vvib00033b6gux9tr8v5%2522%252C%2522activeTabId%2522%253A%2522clwj1vvib00023b6g0eatr4cs%2522%257D%252C%2522clwj1vvib00073b6gi11go318%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clwj1vvib00063b6gd6ehyo2s%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clwj1vvib00073b6gi11go318%2522%252C%2522activeTabId%2522%253A%2522clwj1vvib00063b6gd6ehyo2s%2522%257D%252C%2522clwj1vvib00053b6glfdsldgw%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clwj1vvib00043b6gvkwjt74r%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%255D%252C%2522id%2522%253A%2522clwj1vvib00053b6glfdsldgw%2522%252C%2522activeTabId%2522%253A%2522clwj1vvib00043b6gvkwjt74r%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)
